test/index.js

```diff
@@ -5,19 +5,16 @@ var semver = require('semver');
 var hasDynamicImport = require('../');
 
 test('hasDynamicImport', function (t) {
-	t.equal(typeof hasDynamicImport, 'function', 'is a function');
-
-	var yes = hasDynamicImport();
-	t.equal(typeof yes, 'boolean', 'returns a boolean');
+	t.equal(typeof hasDynamicImport, 'boolean', 'exports a boolean');
 
 	t.test('browser', { todo: true, skip: typeof window === 'undefined' }, function (st) {
 		st.end();
 	});
 
 	t.test('node', { skip: typeof global === 'undefined' }, function (st) {
 		st.equal(
-			yes,
-			semver.satisfies(process.version, '>= 10'),
+			hasDynamicImport,
+			semver.satisfies(process.version, '^12.17 || ^13.2 || >=14'),
```

Comment:
it seems useful to have two exports - one for "the syntax parses" and another for "it actually works".

(I'm fine with the main export being the latter)

Reply:
do you mean something like `require('has-dynamic-import`)` and `require('has-dynamic-import/syntax')`?

should the exports be functions or booleans? i see essentially all other inspect-js packages except has-package-export returns functions.

Reply:
Yea, exactly that. I’d expect the sync one to be a boolean (because the value couldn’t conceptually change later) but the async one should be a function (because you don’t want to kick off the asynchronous check unless it’s needed).

Reply:
finally had some time to update the PR...

`require('has-dynamic-import')` now exports a function that returns a Promise that resolves a boolean indicating if the environment has full dynamic import support.

`require('has-dynamic-import/syntax')` exports a function that returns a boolean indicating if the environment has syntax support for dynamic imports but does not check functional support.

i made both exports return functions for consistency, but can make `has-dynamic-import/syntax` export a boolean instead if you prefer.

I also implemented browser compatible versions of each check.

Reply:
Awesome, thanks! I'll take a look.